Print "verifychain" progress every 1 step if user verifies whole blockchain.

yakitorifoodie opened the pull request:
Currently "verifychain" prints progress every 10 steps. If a user tries to verify the whole blockchain (verifychain 4 0) it will take a lot of time and printing progress every 1 step is very useful.

<!--
*** Please remove the following help text before submitting: ***

Pull requests without a rationale and clear improvement may be closed
immediately.
-->

<!--
Please provide clear motivation for your patch and explain how it improves
Bitcoin Core user experience or Bitcoin Core developer experience
significantly:

* Any test improvements or new tests that improve coverage are always welcome.
* All other changes should have accompanying unit tests (see `src/test/`) or
  functional tests (see `test/`). Contributors should note which tests cover
  modified code. If no tests exist for a region of modified code, new tests
  should accompany the change.
* Bug fixes are most welcome when they come with steps to reproduce or an
  explanation of the potential issue as well as reasoning for the way the bug
  was fixed.
* Features are welcome, but might be rejected due to design or scope issues.
  If a feature is based on a lot of dependencies, contributors should first
  consider building the system outside of Bitcoin Core, if possible.
* Refactoring changes are only accepted if they are required for a feature or
  bug fix or otherwise improve developer experience significantly. For example,
  most "code style" refactoring changes require a thorough explanation why they
  are useful, what downsides they have and why they *significantly* improve
  developer experience or avoid serious programming bugs. Note that code style
  is often a subjective matter. Unless they are explicitly mentioned to be
  preferred in the [developer notes](/doc/developer-notes.md), stylistic code
  changes are usually rejected.
-->

<!--
Bitcoin Core has a thorough review process and even the most trivial change
needs to pass a lot of eyes and requires non-zero or even substantial time
effort to review. There is a huge lack of active reviewers on the project, so
patches often sit for a long time.
-->

yakitorifoodie committed:
Print "verifychain" progress every 1 step.

Currently "verifychain" prints every 10 steps. If a user tries to verify the whole blockchain (verifychain 4 0) it will take a lot of time and printing progress every 1 step is very useful.

MarcoFalke commented on src/validation.cpp:
```diff
@@ -4262,10 +4262,10 @@ bool CVerifyDB::VerifyDB(const CChainParams& chainparams, CCoinsView *coinsview,
         boost::this_thread::interruption_point();
         const int percentageDone = std::max(1, std::min(99, (int)(((double)(::ChainActive().Height() - pindex->nHeight)) / (double)nCheckDepth * (nCheckLevel >= 4 ? 50 : 100))));
-        if (reportDone < percentageDone/10) {
+        if (reportDone < percentageDone/1) {
```
Maybe extract this as a constant that is set to `10` when nCheckDepth is less than 1000 blocks and to `1` otherwise?

laanwj replied:
I agree with @MarcoFalke.

h2o10 replied:
@MarcoFalke  @laanwj   Done. Thank you.

yakitorifoodie committed:
Update validation.cpp

report every 10% step if nCheckDepth is <= 1000 blocks else report every 1% step.

MarcoFalke commented on src/validation.cpp:
```diff
@@ -4258,14 +4258,16 @@ bool CVerifyDB::VerifyDB(const CChainParams& chainparams, CCoinsView *coinsview,
     BlockValidationState state;
     int reportDone = 0;
+    int logStep;
+    logStep = nCheckDepth > 1000 ? 1 : 10;
```
According to the dev notes

maflcko reviewed (commented):
Please squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

yakitorifoodie committed:
Update src/validation.cpp

Co-authored-by: MarcoFalke <falke.marco@gmail.com>

yakitorifoodie committed:
Update validation.cpp

adamjonas commented:
Hi @h2o10, nice to see you contributing for the first time! Pinging you on the squash [requested above](https://github.com/bitcoin/bitcoin/pull/19009#pullrequestreview-418267181). You'll need to do that before this can be merged.

jonatack commented on src/validation.cpp:
```diff
@@ -4258,14 +4258,15 @@ bool CVerifyDB::VerifyDB(const CChainParams& chainparams, CCoinsView *coinsview,
+            // report every 10% step if nCheckDepth is <= 1000 blocks else report every 1% step.
             LogPrintf("[%d%%]...", percentageDone); /* Continued */
-            reportDone = percentageDone/10;
+            reportDone = percentageDone/log_step;
```
suggested change per clang formatting:
```diff
-        if (reportDone < percentageDone/log_step) {
+        if (reportDone < percentageDone / log_step) {
             // report every 10% step if nCheckDepth is <= 1000 blocks else report every 1% step.
             LogPrintf("[%d%%]...", percentageDone); /* Continued */
-            reportDone = percentageDone/log_step;
+            reportDone = percentageDone / log_step;
         }
```

jonatack replied:
```suggestion
            reportDone = percentageDone / log_step;
```

jonatack commented on src/validation.cpp:
```diff
@@ -4258,14 +4258,15 @@ bool CVerifyDB::VerifyDB(const CChainParams& chainparams, CCoinsView *coinsview,
     int nGoodTransactions = 0;
     BlockValidationState state;
     int reportDone = 0;
+    const int log_step{nCheckDepth > 1000 ? 1 : 10};
```
Perhaps place the explanatory comment of `// report every 10% step if nCheckDepth is <= 1000 blocks else report every 1% step` here where you initialize `log_step`.

jonatack commented on src/validation.cpp:
```diff
@@ -4258,14 +4258,15 @@ bool CVerifyDB::VerifyDB(const CChainParams& chainparams, CCoinsView *coinsview,
         const int percentageDone = std::max(1, std::min(99, (int)(((double)(::ChainActive().Height() - pindex->nHeight)) / (double)nCheckDepth * (nCheckLevel >= 4 ? 50 : 100))));
-        if (reportDone < percentageDone/10) {
-            // report every 10% step
+        if (reportDone < percentageDone/log_step) {
```
```suggestion
        if (reportDone < percentageDone / log_step) {
```

jonatack commented on src/validation.cpp:
```diff
@@ -4320,10 +4321,10 @@ bool CVerifyDB::VerifyDB(const CChainParams& chainparams, CCoinsView *coinsview,
             const int percentageDone = std::max(1, std::min(99, 100 - (int)(((double)(::ChainActive().Height() - pindex->nHeight)) / (double)nCheckDepth * 50)));
-            if (reportDone < percentageDone/10) {
-                // report every 10% step
+            if (reportDone < percentageDone/log_step) {
```
```suggestion
            if (reportDone < percentageDone / log_step) {
```

jonatack commented on src/validation.cpp:
```diff
@@ -4320,10 +4321,10 @@ bool CVerifyDB::VerifyDB(const CChainParams& chainparams, CCoinsView *coinsview,
+                // report every 10% step if nCheckDepth is <= 1000 blocks else report every 1% step.
                 LogPrintf("[%d%%]...", percentageDone); /* Continued */
-                reportDone = percentageDone/10;
+                reportDone = percentageDone/log_step;
```
```suggestion
                reportDone = percentageDone / log_step;
```

jonatack reviewed (commented):
Concept ACK

A few formatting nits while retouching to squash the commits.

yakitorifoodie committed:
Update src/validation.cpp

Co-authored-by: Jon Atack <jon@atack.com>

yakitorifoodie committed:
Update src/validation.cpp

Co-authored-by: Jon Atack <jon@atack.com>

yakitorifoodie committed:
Update src/validation.cpp

Co-authored-by: Jon Atack <jon@atack.com>

yakitorifoodie committed:
Update src/validation.cpp

Co-authored-by: Jon Atack <jon@atack.com>

adamjonas commented:
@yakitorifoodie - let me know if you'd like a hand with [squashing your commits]( https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits). I'm ajonas on IRC or you can email me at jonas at chaincode.com if you want to talk through it.

MarcoFalke commented:
Are you still working on this. Otherwise I suggest closing "Up for grabs"

fanquake commented:
This can be re-opened if needed.
